crates/sui-core/src/consensus_handler.rs

```diff
@@ -33,6 +36,10 @@ pub struct ConsensusHandler<T> {
     transaction_manager: Arc<TransactionManager>,
     /// parent_sync_store is needed when determining the next version to assign for shared objects.
     parent_sync_store: T,
+    /// Reputation scores used by consensus adapter that we update, forwarded from consensus
+    low_scoring_authorities: Arc<DashMap<AuthorityName, u64>>,
```

Comment:
I first had this as a trait and followed the pattern of the network connectivity, as you can see in earlier commits, but I removed it for simplicity, especially since the feedback from the last PR was trending towards removing extra components for simplicity. ([@andll](https://github.com/andll)) IMO traits are used when you have different versions of implementations of a function, which we don't have use for here that I see. Maybe a struct though. I'm going to add this to the todo list in the interest of time, and for now I've moved the decision logic into a separate file as a quick and cheap encapsulation.